test/Microsoft.DotNet.ShellShimMaker.Tests/ShellShimMakerTests.cs

```diff
@@ -0,0 +1,153 @@
+
+            shellShimMaker.CreateShim(
+                outputDll.FullName,
+                shellCommandName);
+            var stdOut = ExecuteInShell(shellCommandName);
+
+            stdOut.Should().Contain("Hello World");
+
+            // Tear down
+            shellShimMaker.Remove(shellCommandName);
+        }
+
+        [Fact]
+        public void GivenAnExecutablePathWithExistingSameNameShimItThrows()
+        {
+            var muxer = new Muxer();
+            var pathToPlaceShim = Path.GetDirectoryName(muxer.MuxerPath);
+            var shellCommandName = nameof(ShellShimMakerTests) + Path.GetRandomFileName();
+
+            MakeNameConflictingCommand(pathToPlaceShim, shellCommandName);
+
+            var shellShimMaker = new ShellShimMaker(pathToPlaceShim);
+
+            Action a = () => shellShimMaker.EnsureCommandNameUniqueness(shellCommandName);
+            a.ShouldThrow<GracefulException>()
+                .And.Message
+                .Should().Contain($"Failed to create tool {shellCommandName}, a command with the same name existed");
+
+            // Tear down
+            shellShimMaker.Remove(shellCommandName);
+        }
+
+
+        [Fact]
+        public void GivenAnExecutablePathWithoutExistingSameNameShimItShouldNotThrow()
+        {
+            var muxer = new Muxer();
+            var pathToPlaceShim = Path.GetDirectoryName(muxer.MuxerPath);
+            var shellCommandName = nameof(ShellShimMakerTests) + Path.GetRandomFileName();
+
+            var shellShimMaker = new ShellShimMaker(pathToPlaceShim);
+
+            Action a = () => shellShimMaker.EnsureCommandNameUniqueness(shellCommandName);
+            a.ShouldNotThrow();
+
+            // Tear down
+            shellShimMaker.Remove(shellCommandName);
+        }
+
+        private static void MakeNameConflictingCommand(string pathToPlaceShim, string shellCommandName)
+        {
+            File.WriteAllText(Path.Combine(pathToPlaceShim, shellCommandName), string.Empty);
+        }
+
+
+        private static string ExecuteInShell(string shellCommandName)
+        {
+            string stdOut;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                ExecuteAndCaptureOutputWithAssert(new ProcessStartInfo
+                {
+                    FileName = "CMD.exe",
+                    Arguments = $"/C {shellCommandName}",
+                    UseShellExecute = false
+                }, out stdOut);
+            }
+            else
+            {
+                ExecuteAndCaptureOutputWithAssert(new ProcessStartInfo
+                {
+                    FileName = "sh",
+                    Arguments = $"-c {shellCommandName}",
+                    UseShellExecute = false
+                }, out stdOut);
+            }
+
+            return stdOut ?? "";
+        }
+
+        private static FileInfo MakeHelloWorldExecutableDll()
+        {
+            const string target = "netcoreapp2.1";
```

Comment:
We have no TestAsset today that is a simple console application that you could re-use?

Reply:
I am actually reusing the existing hello world project test asset. I just compile it here.